src/compile/mark/valueref.ts

```diff
@@ -135,6 +135,16 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   return defaultRef;
 }
 
+export function tooltip(def: FieldDef | ValueDef<string>, config: Config): VgValueRef {
```

Comment:
This new method doesn't deal with `timeUnit`, `bin`, and `format` correctly yet. 

Instead of creating new helper method here, I believe you can modify `text()` in `mixins.ts`  to accept a second optional parameter 

```
export function text(model: UnitModel, vgChannel: 'text' | 'tooltip' = 'text') {
   ...
}
```

You can then replace `...mixins.tooltip(model)` with `...mixins.text(model, 'text')` in all mark compilers.

With these minor modifications, I believe it should work correctly without a need for new methods in both `valueref.ts` and `mixins.ts`.  (Of course, please try to test some of these different cases.)